pkg/proxy/usergating/controller.go

```diff
@@ -0,0 +1,200 @@
+/*
+Copyright 2022 The KCP Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package usergating
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	kcpcache "github.com/kcp-dev/apimachinery/pkg/cache"
+	"github.com/kcp-dev/client-go/informers"
+	tenancyv1alpha1 "github.com/kcp-dev/kcp/pkg/apis/tenancy/v1alpha1"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+	"k8s.io/klog/v2"
+
+	"github.com/redhat-cps/front-proxy/pkg/proxy/usergating/ruleset"
+)
+
+const (
+	controllerName = "kcp-user-gating"
+	resyncPeriod   = 10 * time.Minute
+
+	UserGateNamespace = "kcp-system"
+	UserGateResource  = "usergate.kcp.dev"
```

Comment:
Should we add flags for the namespace and secret name as [discussed upstream](https://github.com/kcp-dev/kcp/pull/2310#discussion_r1015657792)?

Not a blocker but it seems like it might be a mitigation for the reading-root-secrets concern if we can ensure RBAC limits access allowed from the proxy?

Reply:
Yep, will do.